nsv13/game/custom_airlocks.dm

```diff
@@ -3,10 +3,101 @@
 	icon = 'nsv13/icons/obj/machinery/doors/standard.dmi'
 	desc = "A durasteel bulkhead which opens and closes. Hope you're good at hatch hopping"
 	icon_state = "closed"
+	overlays_file = 'nsv13/icons/obj/machinery/doors/overlays.dmi'
 //	doorOpen = 'DS13/sound/effects/tng_airlock.ogg'
 //	doorClose = 'DS13/sound/effects/tng_airlock.ogg'
 //	doorDeni = 'DS13/sound/effects/denybeep.ogg'
 
+/obj/machinery/door/airlock/ship/Initialize()
+	. = ..()
+	set_smooth_dir()
+
+
+/*
+
+This proc looks like it makes no sense. Bear with me...
+
+
+Assuming:
+
+X = wall
+[] = airlock:
+
+
+X
+[]
+X
+odir = SOUTH / NORTH. We need it to face sideways so you can get through
+
+X[]X
+odir = EAST / WEST. We need it to face forwards so you can get through
+
+
+*/
+
+/obj/machinery/door/airlock/proc/set_smooth_dir() //I fucking hate this code and so should you :)
+//	for(var/atom/obstacle in view(1, src)) //Ghetto ass icon smooth
+	var/odir = 0
+	var/atom/found = null
+	var/turf/north = get_turf(get_step(src,NORTH))
+	if(north.density)
+		found = north
+		odir = NORTH
+	var/turf/south = get_turf(get_step(src,SOUTH))
+	if(south.density)
+		found = south
+		odir = SOUTH
+	var/turf/east = get_turf(get_step(src,EAST))
+	if(east.density)
+		found = east
+		odir = EAST
+	var/turf/west = get_turf(get_step(src,WEST))
+	if(west.density)
+		found = west
+		odir = WEST
+	if(!found)
+		for(var/atom/foo in get_step(src,NORTH))
+			if(foo?.density)
+				found = foo
+				odir = NORTH
+				break
+		for(var/atom/foo in get_step(src,SOUTH))
+			if(foo?.density)
+				found = foo
+				odir = SOUTH
+				break
+		for(var/atom/foo in get_step(src,EAST))
+			if(foo?.density)
+				found = foo
+				odir = EAST
+				break
+		for(var/atom/foo in get_step(src,WEST))
+			if(foo?.density)
+				found = foo
+				odir = WEST
+				break
+	if(odir == NORTH || odir == SOUTH)
+		dir = EAST
+	else
+		dir = SOUTH
+	return odir
+
+/obj/machinery/door/airlock/ship/station
+	name = "Standard airlock"
+	icon = 'nsv13/icons/obj/machinery/doors/station.dmi'
+
+/obj/machinery/door/airlock/ship/station/mining
+	name = "Mining airlock"
+	color = "#b3784d"
```

Comment:
No. Bad. Don't do this. Don't just throw a color over the sprite, it looks awful ingame because it also fucks with the color of overlays and generally looks nasty. Just give them a damn color stripe or some shit.

Reply:
Looks alright to me chief 
https://cdn.discordapp.com/attachments/602204840180842497/646439400418443274/unknown.png